### PR TITLE
Generalize detail config #406

### LINKED ISSUE #406: Generalize detail config
#### Need
We need a single, re-usable detail config for configuring entity detail pages and export pages. Specifically, we would like to generalize the existing detail config [here](https://github.com/clevercanary/data-browser/blob/main/explorer/app/config/common/entities.ts#L85) so it can be re-used for the export pages (e.g. [here](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=1012%3A15549), [here](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=1012%3A15585) and [here](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=1012%3A16977)). 

#### Approach
* Refactor the detail-specific config to a `BackPage` naming convention (`BackPageConfig`, `BackPageTabConfig`).
* Update tabs to be optional. I think I would prefer the tabs to be optional rather than a blank/empty object here and then "detect and reject" tab functionality as early as possible (to avoid optional chaining throughout). Let me know if this isn't clear!

#### Definition of Done
* Corresponding TSDocs updated.
* HCA project detail pages render unchanged.
* AnVIL dataset detail pages render unchanged.

### PR DESCRIPTION
**Pull request by BruceRodrigues**
### Ticket
Closes #406

### Reviewers
@.

### Changes
- Renaming Tabs config related to BackPage
- Making optional

### Definition of Done (from ticket)
- Corresponding TSDocs updated.
- HCA project detail pages render unchanged.
- AnVIL dataset detail pages render unchanged.

### CODE REVIEW

File: explorer/app/hooks/useCurrentBackPage.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  const currentTab = useMemo(
+    () =>
+      !currentEntity.backPage.tabs
+        ? ({
+            label: "",
+            mainColumn: [],
+            route: "",
+            sideColumn: [],
+          } as BackPageConfig)
```
Comment (BruceRodrigues) [inline explorer/app/hooks/useCurrentBackPage.ts:35]: This will change for https://github.com/clevercanary/data-browser/issues/405

**Comment by MillenniumFalconMechanic** [comment]
Hi @BruceRodrigues, my apologies but I don't think I clearly communicated the task for #406.

I am only expecting the model used by the detail page configuration to be renamed (and not all `detail`-related models/configuration keys). Specifically, I am only expecting a change in naming of `DetailConfig` and `DetailTabConfig` to `BackPageConfig` and `BackPageTabConfig`. So for example, the `EntityConfig` detail page key would remain as `detail` but use the new `BackPageConfig` model:

```
export interface EntityConfig {
    ...
    detail: BackPageConfig
    ...
}
```

Let me know if I have missed something here or if this isn't clear. If possible, would you mind taking another crack at this task?

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Hi Bruce, I have added a comment on this PR. Would you mind taking another look at this?

**Comment by BruceRodrigues** [comment]
@MillenniumFalconMechanic I have opened a new PR for this and will close this one. https://github.com/clevercanary/data-browser/pull/445
For this new PR I'm not addressing the tabs as optional, bc if I do I'll need to add some workaround just to change it when I do https://github.com/clevercanary/data-browser/issues/405.
Is that ok?

**Comment by MillenniumFalconMechanic** [comment]
Sounds good, thanks Bruce!

**Pull request closed without merging**